core/domain/wipeout_service.py

```diff
@@ -396,13 +388,23 @@ def remove_user_from_activities_with_associated_rights_models(
         rights_manager.deassign_role_for_exploration(
             user_services.get_system_user(), exp_id, user_id)
 
-    if use_user_subscriptions_ids:
-        subscribed_collection_summaries = (
-            collection_services.get_collection_summaries_subscribed_to(user_id))
-    else:
-        subscribed_collection_summaries = (
-            collection_services.get_collection_summaries_where_user_has_role(
-                user_id))
+    # The summary model is hard-deleted when the exploration is only marked
+    # as deleted. Thus we need to retrieve the exploration in a different way
+    # and hard-delete it.
```

Comment:
I'm unable to understand this comment, I don't see anything related to summary below. Also, what's the "different way" here? [I mean what was the other way?]

Reply:
Done.